### PR TITLE
Contract to product repo

### PR DESCRIPTION
**Pull request by AlexSJ1990**

### CODE REVIEW

File: src/main/java/com/hedvig/contract/shared/models/ContractState.kt [hunk 1]
```diff
@@ -59,6 +64,15 @@ abstract class ContractState<TAgreement : AgreementState>(
     this.agreements += agreement
   }
 
+  val currentOrLastAgreement: TAgreement
+    get() = if (hasPendingAgreement) pendingAgreement!!
+    else try {
+      val today = LocalDate.now()
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/shared/models/ContractState.kt]: This should probably be moved to Stockholm

File: src/main/java/com/hedvig/contract/shared/models/repositories/ContractToProductRepository.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  fun getCurrentOrLastInsurance(memberId: String): ProductEntity {
+    val currentOrLastAgreement = contractStateRepository.findByHolderMemberId(memberId).first().currentOrLastAgreement
+    return legacyUtils.toProductEntity(currentOrLastAgreement)
+  }
+
+  fun findUserById(memberId: String): UserEntity {
+    return UserEntity(memberService.getMember(memberId))
+  }
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/shared/models/repositories/ContractToProductRepository.kt]: We coult move this but this is fine I think

File: src/main/java/com/hedvig/contract/util/LegacyUtils.kt [hunk 3]
```diff
@@ -1,17 +1,75 @@
...
+      product.personsInHouseHold = agreementState.coInsured.size + 1
+      product.houseType = Product.ProductTypes.valueOf(agreementState.lineOfBusiness.name)
+      product.perils = insuranceBillingService.getPerilsAsMap(Product.ProductTypes.valueOf(agreementState.lineOfBusiness.name))
+    }
+
+    if(agreementState is HouseAgreementState) {
+      product.livingSpace = agreementState.squareMeters.toFloat()
+      product.personsInHouseHold = agreementState.coInsured.size + 1
+      product.houseType = Product.ProductTypes.HOUSE
+      product.perils = insuranceBillingService.getPerilsAsMap(Product.ProductTypes.HOUSE)
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/util/LegacyUtils.kt]: Can be argued we should takeover this as well

File: src/main/java/com/hedvig/contract/shared/models/repositories/ContractToProductRepository.kt [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import java.util.UUID
+
+@Component
+class ContractToProductRepository(
+  private val legacyUtils: LegacyUtils,
+  private val memberService: MemberService,
+  private val contractStateRepository: ContractStateRepository<AgreementState, ContractState<AgreementState>>,
+  private val agreementStateRepository: AgreementStateRepository<AgreementState>,
+  private val perilsManager: PerilsManager
+) {
```
Comment (johantjellden) [inline src/main/java/com/hedvig/contract/shared/models/repositories/ContractToProductRepository.kt]: Should this class not extend from the old ProductEntityRepository?

**Pull request merged**